tests/CMakeLists.txt

```diff
@@ -33,6 +33,7 @@ target_link_libraries(test_library_grib2 PUBLIC sp::sp_d)
 target_link_libraries(test_library_grib2 PUBLIC ip::ip_d)
 
 add_executable(test_gdswzd_grib2 test_gdswzd_grib2.c)
+set_property(TARGET test_gdswzd_grib2 PROPERTY LINKER_LANGUAGE C)
```

Comment:
The errors kept telling me that it was trying to build a Fortran executable but clearly this C with int main() in the source.  So the solution is to explicitly tell CMake this is a C executable